envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotUpdater.kt

```diff
@@ -15,20 +15,28 @@ class SnapshotUpdater(
     private val properties: SnapshotProperties,
     private val scheduler: Scheduler,
     private val onGroupAdded: Flux<out List<Group>>,
-    private val meterRegistry: MeterRegistry
+    private val meterRegistry: MeterRegistry,
+    envoyHttpFilters: EnvoyHttpFilters = EnvoyHttpFilters.emptyFilters,
+    serviceTagFilter: ServiceTagFilter = ServiceTagFilter(properties.routing.serviceTags)
 ) {
     companion object {
         private val logger by logger()
     }
 
     private val versions = SnapshotsVersions()
     private val snapshotFactory = EnvoySnapshotFactory(
-            ingressRoutesFactory = EnvoyIngressRoutesFactory(properties),
-            egressRoutesFactory = EnvoyEgressRoutesFactory(properties),
-            clustersFactory = EnvoyClustersFactory(properties),
-            snapshotsVersions = versions,
-            properties = properties,
-            meterRegistry = meterRegistry
+        ingressRoutesFactory = EnvoyIngressRoutesFactory(properties),
+        egressRoutesFactory = EnvoyEgressRoutesFactory(properties),
+        clustersFactory = EnvoyClustersFactory(properties),
+        listenersFactory = EnvoyListenersFactory(
+                properties,
+                envoyHttpFilters.ingressFilters,
```

Comment:
I have a gut feeling you're breaking Law of Demeter here. Why not just pass in the `EnvoyHttpFilters` entirely?